com.unity.shadergraph/Editor/Generation/Processors/PropertyCollector.cs

```diff
@@ -124,8 +109,24 @@ private List<HLSLProperty> BuildHLSLPropertyList()
             if (m_HLSLProperties == null)
             {
                 m_HLSLProperties = new List<HLSLProperty>();
+                var dict = new Dictionary<string, int>();
```

Comment:
Here we use a dictionary to detect and remove duplicate names.
This lets us find multiple instances of "SamplerState_Linear_Repeat" for example, and merge them.

The Error spew on colliding names is the only thing that's a bit iffy here -- possible this might generate some errors where there were previously none.. though duplicate property names are mighty suspicious and probably should be an error....